meson.build

```diff
@@ -0,0 +1,304 @@
+#endif
+'''
+
+machine = get_option('machine')
+if machine == 'auto'
+  if cc.sizeof('size_t') == 8
+    if conf.get('HAVE_GCC_ASM_FOR_X64') == 1
+      machine = 'x64'
+    elif conf.get('HAVE_UINT128_T') == 1
+      machine = 'uint128'
+    else
+      machine = 'ansi64'
+    endif
+  else
+    if cc.get_id() == 'ccomp'
+      machine = 'ansi-legacy'
+    else
+      machine = 'ansi32'
+    endif
+    if conf.get('HAVE_GCC_ASM_FOR_X87') == 1
+      if cc.get_id() == 'gcc' or cc.get_id() == 'clang'
+        if target_machine.system() != 'darwin'
+          machine = 'ppro'
+        endif
+      endif
+    endif
+  endif
+endif
+
+if machine == 'x64'
+  add_project_arguments('-DCONFIG_64', '-DASM', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_64)
+
+elif machine == 'uint128'
+  add_project_arguments('-DCONFIG_64', '-DANSI', '-DHAVE_UINT128_T', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_64)
+
+elif machine == 'ansi64'
+  add_project_arguments('-DCONFIG_64', '-DANSI', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_64)
+
+elif machine == 'full-coverage'
+  add_project_arguments('-DCONFIG_32', '-DANSI', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_fullcov)
+
+elif machine == 'ppro'
+  add_project_arguments('-DCONFIG_32', '-DPPRO', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32)
+
+elif machine == 'ansi32'
+  add_project_arguments('-DCONFIG_32', '-DANSI', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32)
+
+elif machine == 'ansi-legacy'
+  add_project_arguments('-DCONFIG_32', '-DANSI', '-DLEGACY_COMPILER', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_32_legacy)
+
+elif machine == 'universal'
+  add_project_arguments('-DUNIVERSAL', language : 'c')
+  conf.set('MPD_HEADER_CONFIG', mpd_header_config_universal)
+endif
+
+mpdecimal_h = configure_file(
+  input : 'libmpdec/mpdecimal.h.in',
+  output : 'mpdecimal.h',
+  configuration : conf)
+
+configure_file(
+  input : 'config.h.meson',
+  output : 'config.h',
+  configuration : conf)
+
+libmpdec_sources = files(
+  'libmpdec/basearith.c',
+  'libmpdec/context.c',
+  'libmpdec/constants.c',
+  'libmpdec/convolute.c',
+  'libmpdec/crt.c',
+  'libmpdec/mpdecimal.c',
+  'libmpdec/mpsignal.c',
+  'libmpdec/difradix2.c',
+  'libmpdec/fnt.c',
+  'libmpdec/fourstep.c',
+  'libmpdec/io.c',
+  'libmpdec/mpalloc.c',
+  'libmpdec/numbertheory.c',
+  'libmpdec/sixstep.c',
+  'libmpdec/transpose.c',
+)
+
+libmpdec = library(
+  'mpdec',
+  libmpdec_sources,
+  version : library_version,
+  darwin_versions : [compat_version, library_version],
+  c_args : ['-fPIC'])
+
+libmpdecpp_sources = files(
+  'libmpdec++/decimal.cc',
+)
+
+libmpdecpp = library(
+  'mpdec++',
+  libmpdecpp_sources, mpdecimal_h,
+  version : library_version,
+  darwin_versions : [compat_version, library_version],
+  link_args : ['-Wl,-undefined,dynamic_lookup'],  
+  cpp_args : ['-fPIC'])
```

Comment:
This should  not use `-fPIC` but instead something like:

```
override_options: ['b_staticpic=true']
```

Reply:
Thank you for the review. While directly passing `-fPIC` is probably not the right thing, I think `override_options: ['b_staticpic=true']` is not either. The original Makefile uses `-fPIC` only for the shared library.

Reply:
In that case you don't need to do anything. Meson does exactly that by default already.

Reply:
Ah! Nice. Fixed